From 0139fb45b852cd170ab3e6b10c76e31efcec1116 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 15:30:11 -0400
Subject: [block] mtip32xx: Use pci_enable_msi() instead of pci_enable_msi_range()

Message-id: <1402673421-17117-9-git-send-email-snitzer@redhat.com>
Patchwork-id: 83913
O-Subject: [RHEL7.1 PATCH 08/18] mtip32xx: Use pci_enable_msi() instead of pci_enable_msi_range()
Bugzilla: 1105703
RH-Acked-by: David Milburn <dmilburn@redhat.com>

BZ: 1105703

Upstream commit c94efe36e283f2837dd4a21eab4b3f8492e3f325
Author: Alexander Gordeev <agordeev@redhat.com>
Date:   Tue Feb 25 22:33:18 2014 +0100

    mtip32xx: Use pci_enable_msi() instead of pci_enable_msi_range()

    Commit "mtip32xx: Use pci_enable_msix_range() instead of
    pci_enable_msix()" was unnecessary, since pci_enable_msi()
    function is not deprecated and is still preferable for
    enabling the single MSI mode. This update reverts usage of
    pci_enable_msi() function.

    Besides, the changelog for that commit was bogus, since
    mtip32xx driver uses MSI interrupt, not MSI-X.

    Cc: Jens Axboe <axboe@kernel.dk>
    Cc: Asai Thambi S P <asamymuthupa@micron.com>
    Cc: linux-pci@vger.kernel.org

    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/mtip32xx/mtip32xx.c b/drivers/block/mtip32xx/mtip32xx.c
index 643773b..c04a82d 100644
--- a/drivers/block/mtip32xx/mtip32xx.c
+++ b/drivers/block/mtip32xx/mtip32xx.c
@@ -4619,8 +4619,8 @@ static int mtip_pci_probe(struct pci_dev *pdev,
  INIT_WORK(&dd->work[7].work, mtip_workq_sdbf7);
 
  pci_set_master(pdev);
- rv = pci_enable_msi_range(pdev, 1, 1);
- if (rv < 0) {
+ rv = pci_enable_msi(pdev);
+ if (rv) {
   dev_warn(&pdev->dev,
    "Unable to enable MSI interrupt.\n");
   goto msi_initialize_err;
-- 
1.7.1